akka-stream/src/main/scala/akka/stream/scaladsl/Hub.scala

```diff
@@ -547,24 +602,9 @@ private[akka] class BroadcastHub[T](bufferSize: Int)
                 n += 1
               }
               head = finalOffset
-              if (!hasBeenPulled(in)) pull(in)
+              tryPull()
```

Comment:
I think this is needed, if we set the `startAfterNrOfConsumers` to `2` and register `1`  consumer and then unregister it, but not have a test to cover this, can add one.